tests/python/test_matrix.py

```diff
@@ -18,6 +18,23 @@ def test_python_scope_vector_operations():
         assert np.allclose(c.to_numpy(), ops(a, b))
 
 
+@ti.host_arch_only
+def test_taichi_scope_vector_operations_with_global_vectors():
+    for ops in vector_operation_types:
+        ti.reset()
```

Comment:
Thanks, I'm not 100% sure here either, maybe others can help confirm. What I wanted is to destroy all variables and kernels after each loop to avoid `No new variables can be declared after kernel invocations or Python-scope tensor accesses` exceptions.

Reply:
FYI `ti.host_arch_only` or `ti.all_archs` already do `ti.reset` for you.